.circleci/config.yml

```diff
@@ -11,16 +11,16 @@ commands:
             git config --global user.name "Your Name"
             echo "export PATH=/home/circleci/.go_workspace/src/github.com/argoproj/argo-cd/hack:\$PATH" | tee -a $BASH_ENV
             echo "export GIT_ASKPASS=git-ask-pass.sh" | tee -a $BASH_ENV
-  dep_ensure:
+  mod_vendor:
```

Comment:
I am not sure it is a good idea to use mod vendor in ci, it makes it different than the actual release. And it is ignored unless we pass it in `go build` and `go test` with `-mod=vendor`

Reply:
Personally I prefer checking in ./vendor to my repos, but the status quo in argo-cd is to download dependencies in CI (it used to use `dep ensure`, so I switched to using `go mod -vendor`.)

However, IIRC this doesn't quite work, as mod vendor doesn't download the entire dependency, only the transitively depended upon go code in the dependencies. This means that *.proto files, for example, aren't included in the download.

https://github.com/golang/go/issues/26366

This means `go mod -vendor` and `dep ensure` behave differently. The `vend` tool (https://github.com/nomad-software/vend) should be able to help.